haystack/core/pipeline/base.py

```diff
@@ -1114,18 +1114,22 @@ def _write_component_outputs(
             if receiver_name not in inputs:
                 inputs[receiver_name] = {}
 
-            # If we have a non-variadic or a greedy variadic receiver socket, we can just overwrite any inputs
-            # that might already exist (to be reconsidered but mirrors current behavior).
-            if not is_socket_lazy_variadic(receiver_socket):
-                inputs[receiver_name][receiver_socket.name] = [{"sender": component_name, "value": value}]
-
-            # If the receiver socket is lazy variadic, and it already has an input, we need to append the new input.
-            # Lazy variadic sockets can collect multiple inputs.
+            if is_socket_lazy_variadic(receiver_socket):
```

Comment:
Let's keep some comments to make the code easier to understand
```suggestion
            if is_socket_lazy_variadic(receiver_socket):
                # If the receiver socket is lazy variadic, we append the new input.
                # Lazy variadic sockets can collect multiple inputs.
```